spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulServiceRegistry.java

```diff
@@ -68,7 +68,7 @@ public void register(ConsulRegistration reg) {
 			NewService service = reg.getService();
 			if (this.heartbeatProperties.isEnabled() && this.ttlScheduler != null && service.getCheck() != null
 					&& service.getCheck().getTtl() != null) {
-				this.ttlScheduler.add(reg.getInstanceId());
+				this.ttlScheduler.add(reg.getService());
```

Comment:
Why this change?

Reply:
`ConsulClient.agentServiceRegister` requires `NewService` instance rather than just a service id.